component/common/src/main/java/org/exoplatform/social/common/xmlprocessor/filters/OpenLinkNewTabFilterPlugin.java

```diff
@@ -53,9 +55,16 @@ private String filterLinkTags(String xmlString){
   }
 
   private void nodeFilter(Node currentNode) {
+    String currentDomain = CommonsUtils.getCurrentDomain();
     if ("a".equalsIgnoreCase(currentNode.getTitle())) {
       String target = currentNode.getAttributes().get("target");
-      if (target == null || !"_blank".equalsIgnoreCase(target)) {
+      if (target != null && "_blank".equalsIgnoreCase(target) && currentNode.getAttributes().get("href").indexOf(currentDomain) == -1 ) {
+        currentNode.getAttributes().put("target", "_blank");
+      } if (target != null && "_self".equalsIgnoreCase(target) && currentNode.getAttributes().get("href").indexOf(currentDomain) != -1) {
+        currentNode.getAttributes().put("target", "_self");
+      } if (target != null && currentNode.getAttributes().get("href").indexOf(currentDomain) != -1) {
+        currentNode.getAttributes().put("target", "_self");
```

Comment:
Can we simplify those checks in a single one ?

Reply:
done!